### ISSUE TITLE
Cable for Sensirion SFM3300-D

### ISSUE THREAD

**Issue by fpdotmonkey** [helpfulengineering/project-tetra#67]

**Comment by fpdotmonkey** [comment]
https://www.mill-max.com/products/new/rugged-4mm-pitch-spring-loaded-connectors

**Comment by fpdotmonkey** [comment]
https://www.nxp.com/products/interfaces/ic-spi-serial-interface-devices/ic-bus-repeaters-hubs-extenders/i2c-bus-extender:P82B715

**Comment by mechatroniks-git** [comment]
Attempted to use https://www.digikey.com/product-detail/en/avx-corporation/009155004201006/478-4684-1-ND/1687183
Contacts slip off the pad when compressed. 
![image](https://user-images.githubusercontent.com/57600622/88707858-b01d4900-d0c7-11ea-850b-953e6b17483a.png)

**Comment by mechatroniks-git** [comment]
This Samtec part is better, but interferes: https://www.digikey.com/product-detail/en/samtec-inc/SIB-104-02-F-S-LC/SAM10715-ND/2664704

![image](https://user-images.githubusercontent.com/57600622/88709105-836a3100-d0c9-11ea-9144-60277146040d.png)

**Comment by mechatroniks-git** [comment]
Going with this part should work: https://www.digikey.com/products/en?keywords=821-22-004-10-002101

![image](https://user-images.githubusercontent.com/57600622/88741305-2768d200-d0f4-11ea-85ce-8296d87bc28a.png)

**Comment by fpdotmonkey** [comment]
Seems like this tab doesnt extend out far enough![image](https://user-images.githubusercontent.com/3139615/88742341-d3132180-d0f6-11ea-807f-610bd26ebb62.jpeg)

**Comment by Burhan-Q** [comment]
@fpdotmonkey Just curious, did you try clipping it from both directions? Happy to extend that tab, let me know what Δ you're seeing as needed.

![image](https://user-images.githubusercontent.com/62214284/88744994-e59d5300-d116-11ea-9c5f-7e709855c555.png)

**Comment by mechatroniks-git** [comment]
@hydronics2, here is the PCB DXF
[Sensirion PCB.zip](https://github.com/tetrabiodistributed/project-tetra/files/4996695/Sensirion.PCB.zip)

**Comment by mechatroniks-git** [comment]
The step file: 
[Sensirion PCB Assembly Mockup.zip](https://github.com/tetrabiodistributed/project-tetra/files/4996702/Sensirion.PCB.Assembly.Mockup.zip)

**Comment by fpdotmonkey** [comment]
@Burhan-Q looking at it closer, I think the issue is actually the fillet on the inside corner of the part stopping it from fully engaging.  That said, it is clipping on and staying affixed. ![image](https://user-images.githubusercontent.com/3139615/88855653-0d88c700-d1a8-11ea-9a4e-b8656c59be60.jpeg)

**Comment by fpdotmonkey** [comment]
It could also be a print quality issue.

**Comment by Burhan-Q** [comment]
@fpdotmonkey I can make that a sharp corner if you'd like to check for a better fit. Let me know what you'd like to do.

**Comment by fpdotmonkey** [comment]
Sure sounds good

**Comment by fpdotmonkey** [comment]
Also what's the part number on this?

**Comment by hydronics2** [comment]
@mechatroniks-git  you're cutting it kinda close here. I'd recommend moving the connector 1-2 mm outboard or just trimming the PCB 1-2mm. Any input?... I usually let those connectors hang off the board anyway to allow more PCB space... but I don't want to mess with dimensions too much if you're designing a clamp thingy too...
![image](https://user-images.githubusercontent.com/3589948/88960641-ad9e2900-d258-11ea-9c4c-cca69abe0c2c.png)

**Comment by hydronics2** [comment]
this is just a general note to remember for later... The P82B715 level shifts the i2c signals to 5V.
If we end up putting the 2nd chip on the master board, 
be sure it level shifts back down to 3.3V or take other action as needed.

**Comment by hydronics2** [comment]
@mechatroniks-git  do we need jumpers? I figure either the P82B715 is landed or it's not. Cheap small boards easy to switch between those with the P82B715 and those without. right? I can put them on no sweat just thought I'd simplify things as needed.

**Comment by Burhan-Q** [comment]
> Also what's the part number on this?

@fpdotmonkey just now logged it on the [Hardware Parts List](https://docs.google.com/spreadsheets/d/1nCw9LUN7ygv8B8hno6poN-m15rsixNSolVTXhXynrfw/edit#gid=0&range=B29) as P-TET-000027

**Comment by hydronics2** [comment]
@mechatroniks-git  for review
![image](https://user-images.githubusercontent.com/3589948/88974177-14c5d880-d26d-11ea-889c-3ed0ac2a3242.png)

**Comment by hydronics2** [comment]
@mechatroniks-git  for review
![image](https://user-images.githubusercontent.com/3589948/88977097-1776fc80-d272-11ea-8c90-ca24d7a1fc24.png)

**Comment by hydronics2** [comment]
@mechatroniks-git   for review 
![image](https://user-images.githubusercontent.com/3589948/88977238-6886f080-d272-11ea-9cf6-88ca98454173.png)

**Comment by mechatroniks-git** [comment]
@hydronics2 
Can we add a parallel cap on the IC? Datasheet calls for .1 and 1uF. Also, add a pullup on SDA1. Thanks

![image](https://user-images.githubusercontent.com/57600622/88989867-7435df80-d291-11ea-9405-2fa514acc1b4.png)

**Comment by hydronics2** [comment]
@mechatroniks-git I made the changes... why only a pull-up on the SDA line and not an SCL line? .... never mind, I see it on the TI data sheet now.  The crazy thing about the TI data sheet is it shows no pullups on the SCL line!... I think I got everything I need Patrick.. I'll add a few pull-ups and we can either populate them or not.
![image](https://user-images.githubusercontent.com/3589948/89067401-0be40e00-d324-11ea-9815-1d0f248fd165.png)

**Comment by hydronics2** [comment]
@mechatroniks-git @fpdotmonkey @Burhan-Q  I submitted a pull-request. Here are the files. https://github.com/hydronics2/project-tetra/tree/master/02_Development/FlowMeter/Sensirion_connector_board
I estimated the pogo pin distance in the step file. The pogo pins stick out 5mm (0.197") from the board and the stroke is 1.39mm (0.055").  
![image](https://user-images.githubusercontent.com/3589948/89073550-93834a00-d32f-11ea-91be-963e18ac1833.png)

**Comment by fpdotmonkey** [comment]
@Burhan-Q @hydronics2 ping me on slack when you have the board and the clip ready for me to print

**Comment by hydronics2** [comment]
> @Burhan-Q @hydronics2 ping me on slack when you have the board and the clip ready for me to print

I posted a step file tin the link above. Check it out and see what you think.

**Comment by fpdotmonkey** [comment]
@hydronics2 in application, are the header pins in the model you posted going to be replaced by the sprung pins like what Patrick suggested?

**Comment by hydronics2** [comment]
Yes that's right. I put some header pins in their and tried to size them to 5mm to represent the pogo pins that will stick out 5mm from the board according tonl the data sheet I referenced (uncompressed)

**Comment by fpdotmonkey** [comment]
@Burhan-Q so I tried filing out that fillet, and that didn't do anything.  However, I bent it a little so the internal angle on it is a bit smaller, and now the fit is a bit better.

**Comment by fpdotmonkey** [comment]
It basically looks like the cross section picture you sent me, so I think it's good.  Time to add a feature for the terminal block to go through I suppose.

**Comment by Burhan-Q** [comment]
Moved to Blocked. See notes [here](https://github.com/tetrabiodistributed/project-tetra/pull/111#issuecomment-667583686).
PCB revision required to address issue shown in image
@fpdotmonkey @hydronics2

![image](https://user-images.githubusercontent.com/62214284/89110173-0e269500-d416-11ea-94ff-2137876fbb6c.png)

**Comment by hydronics2** [comment]
haha.. that's funny. I didn't move that part I swear.... but I can certainly move it outboard as many mm as necessary... say start with 5mm?

**Comment by mechatroniks-git** [comment]
We have a mystery to solve, last I left it, the connectors were 8.43mm (center to center distance from the row).  Pushing it out 5mm would work, give me the center to center distance, and I can check.

![image](https://user-images.githubusercontent.com/57600622/89197004-e5aab200-d55f-11ea-8c9d-7282005a6488.png)

**Comment by Burhan-Q** [comment]
Perhaps it's the spacing from the front edge of the board to the phoenix connector thru holes? [both measurements captured in Fusion, from 'front face' of board to center of thru hole]

@mechatroniks-git board measurement **{M}**
![image](https://user-images.githubusercontent.com/62214284/89200335-dd616f80-d57d-11ea-8664-a2a1d03473c1.png)

@hydronics2  board measurement **{H}**
![image](https://user-images.githubusercontent.com/62214284/89200341-df2b3300-d57d-11ea-9bef-e14c6341a937.png)

Profile view with phoenix connector visible, looks like there's a difference in phoenix connector position/length. 

@mechatroniks-git **{M}**
![image](https://user-images.githubusercontent.com/62214284/89201496-91afc580-d57f-11ea-93a8-813aa4afd5ff.png)

@hydronics2 **{H}**
![image](https://user-images.githubusercontent.com/62214284/89200740-71cbd200-d57e-11ea-9cc9-dbd8f78d6c13.png)

**Comment by Burhan-Q** [comment]
> @Burhan-Q so I tried filing out that fillet, and that didn't do anything. However, I bent it a little so the internal angle on it is a bit smaller, and now the fit is a bit better.

@fpdotmonkey do you mean that I should decrease the angle of the clip section like this?
![image](https://user-images.githubusercontent.com/62214284/89207876-747ff480-d589-11ea-9a1e-ff57f1939008.png)

**Comment by hydronics2** [comment]
@mechatroniks-git @Burhan-Q  So it looks like the pogo pins are fine but the large connector exact position needs to be resolved. I can put it wherever it needs to be. This is the latest revision... 
![image](https://user-images.githubusercontent.com/3589948/89212106-c0c23900-d577-11ea-9dfb-40974f244fe1.png)

**Comment by Burhan-Q** [comment]
Blocked by #113 

- [ ] adjust position of phoenix connector pins

**Comment by fpdotmonkey** [comment]
@Burhan-Q the design for the clip is good no need to change it

**Comment by Burhan-Q** [comment]
@fpdotmonkey clip updated with cut out for PCB. Ready for testing

[P-TET-000027-0.zip](https://github.com/tetrabiodistributed/project-tetra/files/5029886/P-TET-000027-0.zip)
![image](https://user-images.githubusercontent.com/62214284/89440902-54cc0600-d71a-11ea-8652-2fca1b364fcf.png)

**Comment by Burhan-Q** [comment]
> @fpdotmonkey clip updated with cut out for PCB. Ready for testing

@fpdotmonkey please move to card to Testing bucket on project board and add Testing Label to the Issue once you begin printing.

**Comment by fpdotmonkey** [comment]
First print of the clip didnt turn out so well. The hook that grabs the ledge nearest the PCB broke off while i was removing support material. Im going to try printing it with the PCB face down to see if that helps at all![image](https://user-images.githubusercontent.com/3139615/89466457-be531100-d728-11ea-8901-9acf09aa0a89.jpeg)

**Comment by fpdotmonkey** [comment]
This is how I'm orienting the next print
![image](https://user-images.githubusercontent.com/3139615/89469354-1c362780-d72e-11ea-9b36-c12f9036c70c.png)

**Comment by fpdotmonkey** [comment]
The previous print was just like this one, but on the other outer face of the clip

**Comment by fpdotmonkey** [comment]
Ok so the connector is nominally working!! The clip connects to the sensor perfectly, and the PCB slides right in. That said, i feel like its really easy to break the PCB-side hook while removing support, and the features for retaining the PCB didnt resolve super well in the print. ![image](https://user-images.githubusercontent.com/3139615/89472425-3fb0a080-d735-11ea-8742-13364c7a86fb.jpeg)

**Comment by fpdotmonkey** [comment]
Also, the clip has a good bit of slop along the sensor's axis (~.075").  I'm concerned that this could cause a short between pins.

**Comment by fpdotmonkey** [comment]
It also only takes a hair of pressure to remove the PCB.  Granted we'll see how hard it is to inert and remove when we have the actual board, but I think an iteration before then could be good

**Comment by Burhan-Q** [comment]
> Also, the clip has a good bit of slop along the sensor's axis (~.075"). I'm concerned that this could cause a short between pins.

@fpdotmonkey could you get me two photos of the extremes of the motion you're seeing, with the Sensirion facing the other direction? I'd looking to see how the harness is moving relative to this feature
![image](https://user-images.githubusercontent.com/62214284/89474435-9dec7c00-d753-11ea-8e27-39fbbb9a1f77.png)

**Comment by fpdotmonkey** [comment]
Full right ![image](https://user-images.githubusercontent.com/3139615/89474861-9a4cfb00-d73b-11ea-9350-540557cc492d.jpeg)
Full left ![image](https://user-images.githubusercontent.com/3139615/89474869-a1740900-d73b-11ea-8a33-fa50baafcd0c.jpeg)

**Comment by fpdotmonkey** [comment]
You can most easily see the difference looking at the right edge of the clip.  I measured the displacement to about .075"

**Comment by Burhan-Q** [comment]
@fpdotmonkey What's the inside distance you measure on the Sensirion here?
![image](https://user-images.githubusercontent.com/62214284/89475455-1ce2b400-d756-11ea-9209-a2b70a3e8ec3.png)

**Comment by hydronics2** [comment]
@fpdotmonkey @Burhan-Q  should I order the PCB? 

Also Fletcher, just checking but you have the mating phoenix phoenix connector... maybe do a quick visual inspection to make sure it plugs in; that is,  the female connector can mate with the PCB-male-phoenix connector. make sense?

**Comment by fpdotmonkey** [comment]
@hydronics2 yes order the PCB

**Comment by fpdotmonkey** [comment]
@Burhan-Q that distance is .727" or 18.47mm

**Comment by hydronics2** [comment]
OK, PCB is ordered. thanks, thomas
Don't forget to order the pogo pins.
https://www.digikey.com/product-detail/en/mill-max-manufacturing-corp/821-22-004-10-002101/ED8199-04-ND/1146641

**Comment by Burhan-Q** [comment]
> @Burhan-Q that distance is .727" or 18.47mm

> You can most easily see the difference looking at the right edge of the clip. I measured the displacement to about .075"

So I can tighten this up some if need be. I didn't know the exact spacing so I left 0.9mm clearance overall (which is about half the amount of movement you're seeing). The contact pads (on the Sensirion model) are 2.3 mm wide and have 0.24 mm gap between them, so I'll lower the spacing to 0.5mm and that should avoid any chance of a short between pins.

### Sensirion SFM3300-D model ### 
![image](https://user-images.githubusercontent.com/62214284/89586430-b1f4b400-d80d-11ea-9328-3d5e57d34587.png)

### Clip current width ### 
![image](https://user-images.githubusercontent.com/62214284/89586484-c769de00-d80d-11ea-8663-ccb8640ed28d.png)

**Comment by fpdotmonkey** [comment]
For the print, I used Hatchbox white PLA.  The spool is a few years old; idk if the material quality degrades over time.

For settings, I had the temps set to 225 end/80 plate, the "standard quality" template on Cura, and the retract speed set to 45mm/s

If there's a way to save out the entire settings file, I'd be happy to do that, but a quick search didn't yield me that result.

**Comment by Burhan-Q** [comment]
> For the print, I used Hatchbox white PLA. The spool is a few years old; idk if the material quality degrades over time.
> 
I believe it will if not humidity controlled 

> For settings, I had the temps set to 225 end/80 plate, the "standard quality" template on Cura, and the retract speed set to 45mm/s
> 

Those temps seem quite high (to me), I also use Hatchbox PLA and print at 200/50 w/ default retraction speed

> If there's a way to save out the entire settings file, I'd be happy to do that, but a quick search didn't yield me that result.

Would be one of the 2 CFG files found in this directory (Win10), not all settings are included, my settings for the print are shown at the bottom. You can also pull the settings from the GCODE file as well, I made a [PowerShell script](https://github.com/Burhan-Q/PowerShell_projects/blob/master/read_PrusaSlicer_gcode.ps1) that pulls the settings from a Prusa Slicer GCODE files (which should be mostly similar to the Cura files).

%APPDATA%\cura\

```
[general]
version = 4
name = creality_base_extruder_0 #2_user
definition = creality_ender3

[values]
brim_width = 2
filter_out_tiny_gaps = True
infill_enable_travel_optimization = True
infill_sparse_density = 15
ironing_only_highest_layer = True
retraction_hop_enabled = True
skirt_line_count = 2
speed_infill = 75
speed_print = 60
support_connect_zigzags = False
support_infill_rate = 10
support_tower_maximum_supported_diameter = 1
support_tree_angle = 60
support_tree_branch_diameter = 1
support_tree_branch_distance = 0.01
support_tree_collision_resolution = 0.1
support_wall_count = 0
wall_line_count = 4
wall_overhang_angle = 70
wall_overhang_speed_factor = 80
z_seam_type = sharpest_corner
```
**Additionally:**
adaptive_layers = True
layer_height = 0.2
slicing_tolerance = Middle

**Comment by fpdotmonkey** [comment]
Got the board in, but it's wider than we expected. The one I 3D printed is 18.08mm wide, whereas this one is 18.88mm. We'll need to adjust the clip to handle this as presently it doesn't fit at all. @Burhan-Q ![image](https://user-images.githubusercontent.com/3139615/89951478-5a28c500-dbe0-11ea-97f0-0bc4483d5dcb.jpeg)

**Comment by fpdotmonkey** [comment]
In case it's relevant, the fully extended length of the pogo pins from the top surface of the board is 6.79 mm and the fully retracted length is 5.76mm

**Comment by fpdotmonkey** [comment]
The big green Pheonix connector is 16.14mm wide, 9.21mm deep, and 7.06mm above the top surface of the board

**Comment by fpdotmonkey** [comment]
Here's a drawing of the dimensions of the board from the factory + me soldering the pogo pins and pheonix connector on
[Sensirion Connector Board.pdf](https://github.com/tetrabiodistributed/project-tetra/files/5059935/Sensirion.Connector.Board.pdf)

**Comment by Burhan-Q** [comment]
@fpdotmonkey could you get me the inside dimensions (HxWxD) from the printed harness (height to under the hold downs)? The current model has an inside width of 19.80mm, so it seems there's some significant material shrinkage and knowing what the difference is will be instructive on how to modify the dimensions.

The model I printed measures:
   depth ~ 20.2mm
   height ~1.5mm (overall pcb harness height minus bottom of pcb harness to bottom of hold downs)
   width ~19.5mm

![image](https://user-images.githubusercontent.com/62214284/90034871-29539900-dc8f-11ea-884b-050daa9edb1b.png)

**Comment by fpdotmonkey** [comment]
I'm getting 18.65mm.  I also noticed there was a bit of extra material there from bad printer settings, so I could change my settings to make it better.  But it may also be good to note on the drawing that the slot that the board goes into needs to be clean.

**Comment by Burhan-Q** [comment]
> I'm getting 18.65mm. I also noticed there was a bit of extra material there from bad printer settings, so I could change my settings to make it better. But it may also be good to note on the drawing that the slot that the board goes into needs to be clean.

@fpdotmonkey, since we have essentially the same printer and filament, maybe you could try using my GCODE file to print the same version I did. That way we can narrow down the issue to printer settings or an issue with the printer or filament. I've attached a zip folder with two versions of my GCODE, one with bed mesh leveling, the other without (since I don't know which you use). These were sliced using Cura 4.3 setup for my Ender 3 using Hatchbox PLA. I don't expect the model to come out perfectly, but it should at least allow us to compare more directly the results from your printer and mine.

[40min_P-TET-000027-0.zip](https://github.com/tetrabiodistributed/project-tetra/files/5070047/40min_P-TET-000027-0.zip)

Additionally, I'm open to any other suggestions or ideas to troubleshoot this issue if you or anyone else have some. Please let me know if you have any questions or issues with the attached GCODE, and I'll keep an eye out for updates on your results.

**Comment by fpdotmonkey** [comment]
So I printed a couple, and the widths of the PCB slot are 18.55mm and 17.84mm.

Other than that, I think the diagonal orientation for the part you're doing is a lot better that the flat one I've been doing.  Way less cleanup and so less risk of breaking the part.  There's also virtually no slop along the sensor's axis and the snap onto the sensor is very clear and satisfying.

**Comment by fpdotmonkey** [comment]
_If it fits it ships_![image](https://user-images.githubusercontent.com/3139615/90298899-c60d6680-de48-11ea-85f1-af1d52ac0383.jpeg)

**Comment by fpdotmonkey** [comment]
Things left to do in testing

- [ ] theres a bit of slop now in the sensor axis still. Probably not consequential, but increasing that width a smidge would be nice
- [ ] validate that the electrical connection is good
- [ ] make 4 at once

After all that, I'm ready to say this part passes review

**Comment by fpdotmonkey** [comment]
Oh also, is the intention for assembling this that the PCB and the clip are put together before the clip is attached to the sensor?  If not, then board doesn't do that.  I think the issue could be that there's a little resistor on the sensor that would block the pogo pin from pushing forward.

**Comment by Burhan-Q** [comment]
> Oh also, is the intention for assembling this that the PCB and the clip are put together before the clip is attached to the sensor? If not, then board doesn't do that. I think the issue could be that there's a little resistor on the sensor that would block the pogo pin from pushing forward.

@fpdotmonkey Yes, the aim (as far as I'm aware) would be to put the PCB into the harness and then attach to the sensor. Just to clarify, you're saying that assembling the clip and the PCB control board first, then to the sensor **_is_** possible?

**Comment by Burhan-Q** [comment]
STL file for P-TET-000027-0
[P-TET-000027-0.zip](https://github.com/tetrabiodistributed/project-tetra/files/5079246/P-TET-000027-0.zip)

**Comment by Burhan-Q** [comment]
@fpdotmonkey are the design changes you mentioned critical for this iteration or are they lower priority (pushed out for beta)?

**Comment by fpdotmonkey** [comment]
The design change I suggested isn't critical

**Comment by fpdotmonkey** [comment]
> you're saying that assembling the clip and the PCB control board first, then to the sensor is possible?

Yes this is possible

**Comment by Burhan-Q** [comment]
> The design change I suggested isn't critical
Okay, then I will make a new ticket for these changes and tag them for the beta prototype

@fpdotmonkey Are you willing to act as reviewer for this design since you've got the printed part(s)? You can review the assembly [here](https://a360.co/2DtJcij) on the web.

I will remove myself from the assignment while you're reviewing. Once you have finished your review, comment here with the outcome of your review, then please assign the task to me (and remove your assignment), and finally if the design passes your review, please indicate that the design is ready for approval.

**Comment by fpdotmonkey** [comment]
Review approved :thumbs-up:

**Comment by Burhan-Q** [comment]
@mechatroniks-git would you be willing to approve this part? The STL file is in the [comment ](https://github.com/tetrabiodistributed/project-tetra/issues/67#issuecomment-674431327) above and @fpdotmonkey [commented ](https://github.com/tetrabiodistributed/project-tetra/issues/67#issuecomment-674306939)about noncritical changes for the next version

**Comment by mechatroniks-git** [comment]
P-TET-000027-0 looks good to me. Approved.
